plugin.video.alfa/channels/videolibrary.py

```diff
@@ -1,6 +1,7 @@
 # -*- coding: utf-8 -*-
 
 import os
+import xbmc

```

Comment:
¿Esto no romperá la versión mediaserver?

Reply:
I do not understand the point. I think that "import xbmc" is arelady used in other modules of Alfa, isn't it?

Reply:
Not so much, and all they are broken:
```Traceback (most recent call last):
  File "/home/osmc/bin/alfa/WebSocket.py", line 68, in run
    controller.run(path)
  File "/home/osmc/bin/alfa/platformcode/controllers/html.py", line 50, in run
    launcher.run(item)
  File "/home/osmc/bin/alfa/platformcode/launcher.py", line 108, in run
    exec "itemlist = channelmodule." + item.action + "(item)"
  File "<string>", line 1, in <module>
  File "/home/osmc/bin/alfa/channels/tvsinpagar.py", line 892, in findvideos
    import xbmc
ImportError: No module named xbmc
```
If you don't want to break non Kodi versions (now only Mediaserver, in the past you have Plex too) you can't use import xbmc in common features, like a channel.

Reply:
May be I'm missing a point but making a search inside the files in plugin.video.alfa, there are a lot of results of the string "import xbmc", for example, downloader.py, videolibrarytools.py, etc. Are these modules not used by Mediaserver version?

Reply:
In your "downloader" example you can see that they are in a try..catch and in "videolibrarytools" it is inside of:
    if config.is_xbmc():

Reply:
In your code is very easy to do platform independent. You only want "import xbmc" to performa  sleep.
You can do some versions:
Implement a platform code "sleep" where XMBC does a xbmc.sleep and others a time.sleep()

or simply put in your code:
try:
    import xbmc
    xbmc.sleep
catch:
    time.sleep

or without exceptions
if config.is_xbmc():
import xbmc
xbmc.sleep()
else:
time.sleep

Reply:
Thanks for further interact on this point. I'm starting to understand but what it is still unclear to me is that the is_xbmc() function inside config.py returns always True...

Reply:
Yes, here is always true: https://github.com/alfa-addon/addon/blob/master/plugin.video.alfa/platformcode/config.py#L57

But here is always false: https://github.com/alfa-addon/addon/blob/master/mediaserver/platformcode/config.py#L32

It is normal that "is_xbmc" is always true in XMBC. And false in others (like mediaserver).

Reply:
Thanks for the reply alaquepasa. my initial understanding was that Alfa could be installed over Kodi and Mediserver...now I understood that there are actually two different add-on that maybe share a lot of code as well as the name of the modules but with customized functions. All clear now; I will implement the code you suggested.